datasets/diplomacy_detection/diplomacy_detection.py

```diff
@@ -0,0 +1,117 @@
+}
+"""
+
+_DESCRIPTION = "The Diplomacy dataset contains pairwise conversations annotated by the sender and the receiver for deception (and conversely truthfulness).   The 17,289 messages are gathered from 12 games.  We train baseline and neural models to predict lies."
+
+_HOMEPAGE = "https://sites.google.com/view/qanta/projects/diplomacy"
+
+# _LICENSE = ""
+
+_TEST_DOWNLOAD_URL = "https://github.com/DenisPeskov/2020_acl_diplomacy/raw/master/data/test.jsonl"
+_TRAIN_DOWNLOAD_URL = "https://github.com/DenisPeskov/2020_acl_diplomacy/raw/master/data/train.jsonl"
+_VALIDATION_DOWNLOAD_URL = "https://github.com/DenisPeskov/2020_acl_diplomacy/raw/master/data/validation.jsonl"
+
+
+class DiplomacyDetection(datasets.GeneratorBasedBuilder):
+    """Diplomacy: A Dataset for Deception Detection. """
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "messages": datasets.Value("string"),
+                "sender_labels": datasets.Value("string"),
+                "receiver_labels": datasets.Value("string"),
+                "speakers": datasets.Value("string"),
+                "receivers": datasets.Value("string"),
+                "absolute_message_index": datasets.Value("string"),
+                "relative_message_index": datasets.Value("string"),
+                "seasons": datasets.Value("string"),
+                "years": datasets.Value("string"),
+                "game_score": datasets.Value("string"),
+                "game_score_delta": datasets.Value("string"),
+                "players": datasets.Value("string"),
+                "game_id": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        train_path = dl_manager.download_and_extract(_TRAIN_DOWNLOAD_URL)
+        test_path = dl_manager.download_and_extract(_TEST_DOWNLOAD_URL)
+        validation_path = dl_manager.download_and_extract(_TEST_DOWNLOAD_URL)
```

Comment:
`dl_manager.download_and_extract` should take a dict of URLs as an input, returning a dict of paths to the downloaded and extracted resources. This enables parrallel processing.

Reply:
For example
```python
downloaded_paths = dl_manager.download_and_extract({
    "train": _TRAIN_DOWNLOAD_URL,
    "test": _TEST_DOWNLOAD_URL,
    "validation": _VALIDATION_DOWNLOAD_URL,
)
```
Then you can replace the return statement by
```python
return [
    datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": downloaded_paths["validation"]}),
    datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_paths["train"]}),
    datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": downloaded_paths["test"]}),
]
```